Tests/MathLib/AutoCheckTools.h

```diff
@@ -15,6 +15,58 @@
 
 namespace autocheck
 {
+
+/// The autocheck standard generator for floating point values returns uniform
+/// distributed values. Autocheck uses the size argument of the operator() to
+/// determine the interval for the uniform distribution. This class template
+/// fixes the size to one and transforms a generated random value (within the
+/// interval [-1,1]) linearly to a value in the interval [a,b].
+template <typename T, typename Gen = generator<T>>
+struct IntervalGenerator
+{
+	/// Construtor initializing the slope and the \f$y\f$-intercept deploying
+	/// lower bound \f$a\f$ and upper bound \f$b\f$ of the interval.
+	IntervalGenerator(T a, T b)
+	    : _m((b-a)/2), _n((b+a)/2)
+	{}
+
+	// parameters for the interval mapping [-1,1] -> [a,b],
+	// y = _m * x + _n
+	T _m{1};
+	T _n{0};
+
+	Gen generator;
+
+	using result_type = T;
+
+	result_type intervalMap(T val) const
+	{
+		return _m * val + _n;
+	}
+
+	result_type operator()(std::size_t size = 0)
```

Comment:
The parameter `size` is not used, and that generates really ugly warning messages.
Please comment/remove that parameter. And do you need the default value? If not, the I would opt to remove it.

```
In file included from /home/lehmannc/prog/ogs6/github-chleh/src/Tests/MeshGeoToolsLib/TestGeoMapper.cpp:19:0:
/home/lehmannc/prog/ogs6/github-chleh/src/Tests/MeshGeoToolsLib/../MathLib/AutoCheckTools.h: In Instanziierung von »autocheck::IntervalTupleGenerator<T, Gen>::result_type autocheck::IntervalTupleGenerator<T, Gen>::operator()(std::size_t) [with T = double; Gen = autocheck::IntervalGenerator<double>; autocheck::IntervalTupleGenerator<T, Gen>::result_type = std::__debug::array<double, 3ul>; std::size_t = long unsigned int]«:
/home/lehmannc/prog/ogs6/github-chleh/src/ThirdParty/autocheck/include/autocheck/generator.hpp:28:32:   erfordert durch »T autocheck::generate(std::tuple<_Elements ...>&, size_t, const autocheck::range<0, Is ...>&) [with T = GeoLib::Point; Gens = {autocheck::IntervalTupleGenerator<double, autocheck::IntervalGenerator<double, autocheck::generator<double, void> > >}; int ...Is = {0}; size_t = long unsigned int]«
/home/lehmannc/prog/ogs6/github-chleh/src/ThirdParty/autocheck/include/autocheck/generator.hpp:33:34:   erfordert durch »T autocheck::generate(std::tuple<_Elements ...>&, size_t) [with T = GeoLib::Point; Gens = {autocheck::IntervalTupleGenerator<double, autocheck::IntervalGenerator<double, autocheck::generator<double, void> > >}; size_t = long unsigned int]«
/home/lehmannc/prog/ogs6/github-chleh/src/ThirdParty/autocheck/include/autocheck/generator.hpp:286:48:   erfordert durch »autocheck::cons_generator<T, Gens>::result_type autocheck::cons_generator<T, Gens>::operator()(size_t) [with T = GeoLib::Point; Gens = {autocheck::IntervalTupleGenerator<double, autocheck::IntervalGenerator<double, autocheck::generator<double, void> > >}; autocheck::cons_generator<T, Gens>::result_type = GeoLib::Point; size_t = long unsigned int]«
/home/lehmannc/prog/ogs6/github-chleh/src/ThirdParty/autocheck/include/autocheck/generator_combinators.hpp:86:26:   erfordert durch »autocheck::detail::fixed_size_generator<Gen>::result_type autocheck::detail::fixed_size_generator<Gen>::operator()(size_t) [with Gen = autocheck::cons_generator<GeoLib::Point, autocheck::IntervalTupleGenerator<double, autocheck::IntervalGenerator<double, autocheck::generator<double, void> > > >; autocheck::detail::fixed_size_generator<Gen>::result_type = GeoLib::Point; size_t = long unsigned int]«
/usr/include/c++/5.3.0/bits/stl_algo.h:4337:18:   [ skipping 2 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/home/lehmannc/prog/ogs6/github-chleh/src/ThirdParty/autocheck/include/autocheck/generator_combinators.hpp:86:26:   erfordert durch »autocheck::detail::fixed_size_generator<Gen>::result_type autocheck::detail::fixed_size_generator<Gen>::operator()(size_t) [with Gen = autocheck::list_generator<autocheck::cons_generator<GeoLib::Point, autocheck::IntervalTupleGenerator<double, autocheck::IntervalGenerator<double, autocheck::generator<double, void> > > > >; autocheck::detail::fixed_size_generator<Gen>::result_type = std::__debug::vector<GeoLib::Point>; size_t = long unsigned int]«
/home/lehmannc/prog/ogs6/github-chleh/src/ThirdParty/autocheck/include/autocheck/generator.hpp:28:32:   erfordert durch »T autocheck::generate(std::tuple<_Elements ...>&, size_t, const autocheck::range<0, Is ...>&) [with T = std::tuple<std::__debug::vector<GeoLib::Point, std::allocator<GeoLib::Point> > >; Gens = {autocheck::detail::fixed_size_generator<autocheck::list_generator<autocheck::cons_generator<GeoLib::Point, autocheck::IntervalTupleGenerator<double, autocheck::IntervalGenerator<double, autocheck::generator<double, void> > > > > >}; int ...Is = {0}; size_t = long unsigned int]«
/home/lehmannc/prog/ogs6/github-chleh/src/ThirdParty/autocheck/include/autocheck/generator.hpp:33:34:   erfordert durch »T autocheck::generate(std::tuple<_Elements ...>&, size_t) [with T = std::tuple<std::__debug::vector<GeoLib::Point, std::allocator<GeoLib::Point> > >; Gens = {autocheck::detail::fixed_size_generator<autocheck::list_generator<autocheck::cons_generator<GeoLib::Point, autocheck::IntervalTupleGenerator<double, autocheck::IntervalGenerator<double, autocheck::generator<double, void> > > > > >}; size_t = long unsigned int]«
/home/lehmannc/prog/ogs6/github-chleh/src/ThirdParty/autocheck/include/autocheck/arbitrary.hpp:50:44:   erfordert durch »bool autocheck::arbitrary<Gens>::operator()(autocheck::value<std::tuple<typename autocheck::arbitrary<Gens>::ExpandArgs<Gens>::result_type ...> >&) [with Gens = {autocheck::detail::fixed_size_generator<autocheck::list_generator<autocheck::cons_generator<GeoLib::Point, autocheck::IntervalTupleGenerator<double, autocheck::IntervalGenerator<double, autocheck::generator<double, void> > > > > >}]«
/home/lehmannc/prog/ogs6/github-chleh/src/ThirdParty/autocheck/include/autocheck/check.hpp:39:15:   erfordert durch »void autocheck::check(Property, size_t, Arbitrary, const autocheck::reporter&, autocheck::classifier<Args ...>, bool) [with Args = {std::__debug::vector<GeoLib::Point, std::allocator<GeoLib::Point> >}; Property = MeshGeoToolsLibGeoMapper_PointsOnSurfaceMesh_Test::TestBody()::<lambda(std::__debug::vector<GeoLib::Point>&)>; Arbitrary = autocheck::arbitrary<autocheck::detail::fixed_size_generator<autocheck::list_generator<autocheck::cons_generator<GeoLib::Point, autocheck::IntervalTupleGenerator<double, autocheck::IntervalGenerator<double, autocheck::generator<double, void> > > > > > >; size_t = long unsigned int]«
/home/lehmannc/prog/ogs6/github-chleh/src/Tests/MeshGeoToolsLib/TestGeoMapper.cpp:87:17:   von hier erfordert
/home/lehmannc/prog/ogs6/github-chleh/src/Tests/MeshGeoToolsLib/../MathLib/AutoCheckTools.h:64:44: Warnung: unbenutzter Parameter »size« [-Wunused-parameter]
  result_type operator()(std::size_t size = 0)
```